### ISSUE TITLE
click toggle element twice should set visibility to Off

### ISSUE THREAD

**Issue by rnons** [citizennet/purescript-halogen-select#20]
Thanks for this nice library.

## Environment

* Select **[0.1.12]**

## Current behavior
Currently, `setToggleProps` only have `onFocus` and `onBlur`. Which means clicking toggle element twice won't turn visibility off.

```
setToggleProps = flip (<>)
  [ HE.onFocus    $ HE.input CaptureFocus
  , HE.onKeyDown  $ HE.input Key
  , HE.onBlur $ HE.input_ $ SetVisibility Off
  , HP.tabIndex 0
  ]
```

## Expected behavior
Clicking toggle element twice should turn visibility off. Should add `onClick` handler, and perhaps `onMouseDown/Up` as well

**Comment by thomashoneyman** [comment]
@rnons Thanks for opening an issue! 

You're right, it does seem more natural that clicking a toggle twice should close it. We have the `ToggleVisibility` query available for that, so we could have `mouseDown` events call that query.

Something like this ought to work:

```purescript
setToggleProps = flip (<>)
  [ HE.onFocus $ HE.input CaptureFocus
  , HE.onKeyDown  $ HE.input Key
  , HE.onMouseDown $ HE.input_ ToggleVisibility
  , HE.onBlur $ HE.input_ $ SetVisibility Off
  , HP.tabIndex 0
  ]
```

@whoadave @crcornwell @foresttoney Unless you have other objections, I can implement this along with a toggle-driven example later today or tomorrow.

---

By the way -- if you need this behavior right now, you can always add it to the properties you're augmenting:

```purescript
...
Setters.setToggleProps 
  [ HP.class_ (HH.ClassName "my-class")
  , HE.onMouseDown $ HE.input_ Select.ToggleVisibility ]
...
```

**Comment by thomashoneyman** [comment]
@rnons Should be fixed in #21
